name: ‚ÄÇüì¶‚ÄÇApp‚ÄÇ‚å®Ô∏è‚ÄÇCLI‚ÄÇüêû‚ÄÇBug Report
description: Report problems and errors in PyPackIT's CLI.
body:
  - type: markdown
    attributes:
      value: |-
        <p align="justify">
          Thank you for providing feedback on PyPackIT;
          we greatly appreciate your contribution!
          Before submitting a ticket,
          please take a moment to review and accept our terms and conditions.
        </p>
  - type: markdown
    attributes:
      value: |
        <br>

        ## ü§ù‚ÄÇTerms and Conditions
  - type: markdown
    attributes:
      value: |
        ### ‚õî‚ÄÇChoosing the Right Form

        <p align="justify">
          Please note that this form is specifically for reporting a <b>non-security-related bug in PyPackIT's CLI</b>, i.e. an unexpected behavior or result in PyPackIT's command-line interface.,
          For other types of issues and requests, select the appropriate form
          from the <a href="https://github.com/RepoDynamics/PyPackIT/issues/new/choose">issue template chooser</a>.
          If you are unsure about which form to choose or have another inquiry,
          please open a discussion in our <a href="https://github.com/RepoDynamics/PyPackIT/discussions">discussion forum</a>.
        </p>
  - type: markdown
    attributes:
      value: |
        ### ‚ö†Ô∏è‚ÄÇVerifying the Bug

        <p align="justify">
          Before submitting a report, please make sure that you are following our
          <a href="https://pypackit.repodynamics.com/contribute">Bug Report Guidelines</a> to verify that:
        </p>

        - <p align="justify">The issue you are experiencing is in fact a bug, and not a documented expected behavior.</p>
        - <p align="justify">The bug is reproducible in the latest version of the release, and has not yet been reported.</p>
  - type: markdown
    attributes:
      value: |
        ### ‚ÑπÔ∏è‚ÄÇChoosing the Right Title

        <p align="justify">Please write a concise and descriptive title and
        use relevant keywords to help others find the issue.</p>
  - type: markdown
    attributes:
      value: '### ‚úÖ‚ÄÇAgreements'
  - type: checkboxes
    id: terms
    attributes:
      label: 'To submit an issue, please agree to the following terms and conditions:'
      options:
        - label: "I confirm that I am following PyPackIT's [Contribution Guidelines](https://pypackit.repodynamics.com/contribute).<!--
            ISSUE-ID: pkg_cli_fix -->"
          required: true
        - label: I agree to follow PyPackIT's [Code of Conduct](https://github.com/RepoDynamics/PyPackIT/blob/main/CODE_OF_CONDUCT.md).
          required: true
  - type: markdown
    attributes:
      value: '### üßë‚ÄçüíªÔ∏è‚ÄÇCollaboration'
  - type: checkboxes
    id: collab
    attributes:
      label: |
        Thank you again for filling out this issue ticket.
        If you are willing to collaborate on its design and implementation,
        please select the option below.
      options:
        - label: I am willing to collaborate on this issue and submit a pull request.
          required: false
  - type: markdown
    attributes:
      value: |
        <br>

        ## üéõ‚ÄÇEnvironment Specifications
  - type: dropdown
    id: os
    attributes:
      label: Operating System
      description: |
        Please select all operating systems on which you have encountered the bug.
      multiple: true
      options:
        - Linux (arm64)
        - Linux (x64)
        - Windows (x64)
        - macOS (arm64)
        - macOS (x64)
    validations:
      required: true
  - type: dropdown
    id: python_versions
    attributes:
      label: Python Version
      description: |
        Please select all Python versions in which you have encountered the bug.
      multiple: true
      options:
        - '3.13'
        - '3.12'
        - '3.11'
        - '3.10'
      default: 0
    validations:
      required: true
  - type: dropdown
    id: package_manager
    attributes:
      label: Package Manager
      description: |
        Please select the package manager used to download and install PyPackIT.
      multiple: false
      options:
        - conda
        - pip
      default: 0
    validations:
      required: true
  - type: textarea
    id: environment
    attributes:
      label: Environment
      description: |-
        Please provide the details of all installed packages and their exact versions in the environment where you have encountered the bug. Follow the instructions below based on the package manager you have selected:
        1. **`pip`**: Run `pip freeze > requirements.txt` in your terminal within the environment, and copy the contents of the generated `requirements.txt` file here.
        2. **`conda`**: Run `conda env export > environment.yml` in your terminal within the environment (or alternatively `conda env export -n {YOUR_ENVIRONMENT_NAME} > environment.yml`), and copy the contents of the generated `environment.yml` file here.
      render: yaml
    validations:
      required: true
  - type: markdown
    attributes:
      value: |
        <br>

        ## üêû‚ÄÇBug Report
  - type: dropdown
    id: version
    attributes:
      label: Version
      description: |
        Please select all relevant PyPackIT versions.
      multiple: true
      options:
        - 0.0.0
      default: 0
    validations:
      required: true
  - type: dropdown
    id: entrypoint
    attributes:
      label: CLI Entry Point
      description: Please select the relevant CLI entry point.
      multiple: false
      options:
        - PyPackIT
      default: 0
    validations:
      required: true
  - type: textarea
    id: summary
    attributes:
      label: Summary
      description: Please provide a summary of the issue.
    validations:
      required: true
  - type: textarea
    id: test_case
    attributes:
      label: Test Case
      description: |-
        Please provide a [test case](https://docs.pytest.org/en/stable/getting-started.html) that demonstrates the bug:
        - In the `test` function below, replace the value of `command` with the exact command that is causing the problem.
        - Replace the value of `expected_return_code` with the expected return code of the command.
        - Replace the value of `expected_output` with the expected output of the command. Use [`assert` statements](https://docs.pytest.org/en/stable/how-to/assert.html) to specify the expected results.
        - If the command has side effects, add further [`assert` statements](https://docs.pytest.org/en/stable/how-to/assert.html) to specify the expected results.
      value: |
        import subprocess


        def test():
            command = ""
            expected_return_code = 0
            expected_output = b""

            result = subprocess.run(command, stdout=subprocess.PIPE, shell=True)
            assert result.returncode == expected_return_code
            assert result.stdout == expected_output
      render: python
    validations:
      required: true
